Tabnine/src/main/java/com/tabnine/lifecycle/ForceRegistration.kt

```diff
@@ -0,0 +1,194 @@
+package com.tabnine.lifecycle
+
+import com.intellij.notification.Notification
+import com.intellij.notification.NotificationType
+import com.intellij.notification.Notifications
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.ServiceManager
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.util.messages.MessageBusConnection
+import com.tabnineCommon.binary.requests.config.StateResponse
+import com.tabnineCommon.binary.requests.login.LoginRequest
+import com.tabnineCommon.capabilities.CapabilitiesService
+import com.tabnineCommon.capabilities.Capability
+import com.tabnineCommon.capabilities.CapabilityNotifier
+import com.tabnineCommon.general.DependencyContainer
+import com.tabnineCommon.general.StaticConfig.NOTIFICATION_ICON
+import com.tabnineCommon.lifecycle.BinaryStateChangeNotifier
+import com.tabnineCommon.lifecycle.BinaryStateService
+import com.tabnineCommon.lifecycle.PluginInstalled
+import java.awt.BorderLayout
+import javax.swing.JComponent
+import javax.swing.JPanel
+import javax.swing.JTextArea
+
+class ForceRegistration() {
+
+    private val capabilities: MessageBusConnection
+
+    private val binaryState: MessageBusConnection
+    fun start() {
+        println("starting force registration flow")
+        if (CapabilitiesService.getInstance().isCapabilityEnabled(Capability.FORCE_REGISTRATION)) {
+            print("force registration capability is enabled so invoking")
+            invoke()
+        }
+    }
+
+    private fun sendAnalytics(event: String) {
+        // find the code that sends the analytics
```

Comment:
I think we missed implementing this one.

Reply:
yes